201-sql-auditing-server-policy-to-blob-storage/azuredeploy.json

```diff
@@ -46,14 +56,40 @@
       "sku": {
         "name": "Standard_LRS"
       },
-      "kind": "Storage",
-      "properties": {}
-    },
-    {
+      "kind": "StorageV2",
+      "properties": {
+        "networkAcls": {
+          "bypass": "AzureServices",
+          "virtualNetworkRules": [],
+          "ipRules": [],
+          "defaultAction": "[if(parameters('isStorageBehindVnet'), 'Deny', 'Allow')]"
+        }
+      },
+      "resources": [
+        {
+          "condition": "[parameters('isStorageBehindVnet')]",
+          "type": "Microsoft.Storage/storageAccounts/providers/roleAssignments",
```

Comment:
simplify this and just use Microsoft.Authorization/roleAssignments

Reply:
This will fail for a storage account resource. This is the format of role assignment on a resource (storage account). I've followed https://docs.microsoft.com/en-us/azure/role-based-access-control/role-assignments-template#resource